jupyter_server/services/kernelspecs/handlers.py

```diff
@@ -48,6 +48,19 @@ def is_kernelspec_model(spec_dict):
     )
 
 
+def update_kernelspec_model(spec_dict):
+    """Returns the original kernelspec unless template substitutions are available."""
+    model = spec_dict
+    if "resource_dir" in spec_dict:
```

Comment:
Not sure when this will be true since it's asking if the kernelspec _model_ contains the key `"resource_dir"` yet, in the models, `resource_dir` is used to _identify_ the other resources of the kernel (icon, etc.) and not retained as a key/value.  As a result, this logic is probably better suited in `kernelspec_model()` directly.

I don't think this resource_dir-substitution logic applies to `spec_dict`s that satisfy the `is_kernelspec_model()` check because, for those, the water is under the bridge and the resource_dir is not available (nor is it local).  The specs returned from a Gateway server are examples of those.